### PR TITLE
add plugin "Canvas Links"

### PR DESCRIPTION
**Pull request by aqav**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/aqav/obsidian-canvas-links

## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [ ] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by marcusolsson** [comment]
["name": "Canvas View"](https://github.com/aqav/obsidian-canvas-view/blob/973121261cf7a5a109ab09c529bcd8ba8aa186f4/manifest.json#L3)
The name of the plugin could be confused with the actual canvas view. Could you think of a more descriptive name for the plugin?

[await this.app.vault.read(activeFile)](https://github.com/aqav/obsidian-canvas-view/blob/973121261cf7a5a109ab09c529bcd8ba8aa186f4/main.ts#L85)
[Looks like you can use `cachedRead` here?](https://docs.obsidian.md/Plugins/Vault#Read+files) 

[this.app.workspace.detachLeavesOfType(FILE_VIEW);](https://github.com/aqav/obsidian-canvas-view/blob/973121261cf7a5a109ab09c529bcd8ba8aa186f4/main.ts#L32)
Consider reusing the leaf instead of detaching it every time.

**Comment by aqav** [comment]
@marcusolsson 

"[Looks like you can use cachedRead here?](https://docs.obsidian.md/Plugins/Vault#Read+files)"
"Consider reusing the leaf instead of detaching it every time."

For these two, I have committed below to meet them, please ref to 
https://github.com/aqav/obsidian-canvas-view/commit/6ca3673827e6b3c30a3d45e4a566d804e2c7b106 
https://github.com/aqav/obsidian-canvas-view/commit/8cb7ac3c186f39241662da4812619a3b1abf223b

"The name of the plugin could be confused with the actual canvas view. Could you think of a more descriptive name for the plugin?"
Yes, I could change the name for an explicit description, but I am not so creative about naming. How about "Canvas Link View" or "Link View of Canvas", which would be better as your think? I want to ask your advice first before changing it for not to change it twice, If you have any suggestions it is welcome.

**Comment by marcusolsson** [comment]
I like "Canvas Link View" better, or maybe just "Canvas Links"?

**Comment by aqav** [comment]
@marcusolsson 
I prefer the short one, it's more elegant. I appreciate your suggestion. The change has been done, plz check https://github.com/aqav/obsidian-canvas-links

**Review by marcusolsson — APPROVED** [review]

**Comment by liamcain** [comment]
- [getFiles](https://github.com/aqav/obsidian-canvas-links/blob/2b72a6536a7a45e8b9b8683dc9b9e0e786e64421/main.ts#L105) instead of looping through all files in the vault, you're better off just using [`getAbstractFileByPath`](https://github.com/obsidianmd/obsidian-api/blob/902badd38ba907689f0917d7b193f7e33d1284fe/obsidian.d.ts#L3612) and getting the TFiles that way.

File: community-plugins.json [hunk 1]
```diff
@@ -6921,5 +6921,13 @@
     "author": "lusob",
     "description": "A ChatGPT powered chatbot specifically focused on question answering over your Obsidian vault notes.",
     "repo": "lusob/obsidian-brain"
+
+  },
```
Comment (liamcain) [inline community-plugins.json:6925]:
```suggestion
  },
```

**Comment by liamcain** [comment]
@aqav seems like I don't have permissions to rebase your branch for you. Could you rebase and move your entry to the bottom of the plugin list? Then I can merge it. Thanks!

**Comment by aqav** [comment]
@liamcain 

Tks for your suggestion, I have update the code, plz ref https://github.com/aqav/obsidian-canvas-links/commit/d1004f590f5fb023ed4ab2e748b9c78a9668ee2d

The conflict has been resolved, my plugin entry have been moved to the bottom of list. plz check

**Pull request merged**